consensus/hotstuff/verification/combined_verifier_v3.go

```diff
@@ -158,14 +164,14 @@ func (c *CombinedVerifierV3) VerifyQC(signers flow.IdentityList, sigData []byte,
 	for _, signerID := range blockSigData.StakingSigners {
 		identity, ok := signerIdentities[signerID]
 		if !ok {
-			return false, fmt.Errorf("invalid signer identity %v", signerID)
+			return fmt.Errorf("invalid signer identity %v: %w", signerID, model.ErrInvalidSignature)
```

Comment:
## ⚠️ 

You correctly argued above that this case should not occur, and reaching this failure condition indicates an internal bug: https://github.com/onflow/flow-go/blob/d2788f297c099edf033310d70fc66cee3a63ce8b/consensus/hotstuff/verification/combined_verifier_v3.go#L147-L148
I think the same argument applies here.  Hence, I think we should _not_ return a `model.ErrInvalidSignature` here, but an exception instead.

Reply:
Good catch. 

I'm also thinking that maybe we should let the packer return Identity instead, then we can eliminate this case